src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -39,33 +44,47 @@ bool ClusterTaskManager::SchedulePendingTasks() {
                      << task.GetTaskSpecification().TaskId();
       auto placement_resources =
           task.GetTaskSpecification().GetRequiredPlacementResources().GetResourceMap();
+      // This argument is used to set violation, which is an unsupported feature now.
       int64_t _unused;
-      // TODO (Alex): We should distinguish between infeasible tasks and a fully
-      // utilized cluster.
+      bool is_infeasible;
       std::string node_id_string = cluster_resource_scheduler_->GetBestSchedulableNode(
           placement_resources, task.GetTaskSpecification().IsActorCreationTask(),
-          &_unused);
+          &_unused, &is_infeasible);
+
+      // There is no node that has available resources to run the request.
+      // Move on to the next shape.
       if (node_id_string.empty()) {
-        // There is no node that has available resources to run the request.
-        // Move on to the next shape.
-        RAY_LOG(DEBUG) << "No feasible node found for task "
+        RAY_LOG(DEBUG) << "No node found to schedule a task "
                        << task.GetTaskSpecification().TaskId();
+        is_shape_infeasible = is_infeasible;
```

Comment:
What's the point in this `is_infeasible` variable? Why not just directly pass `&is_shape_infeasible` to `GetBestSchedulableNode`?